File: src/test/java/at/fhhagenberg/sqe/di/MockRmiElevatorModule.java
package at.fhhagenberg.sqe.di;

import sqelevator.IElevator;
import com.google.inject.*;

public class MockRmiElevatorModule extends AbstractModule {

    @Override
    protected void configure() {
        bind(Key.get(IElevator.class, RealIElevator.class)).toProvider(RealIElevatorMockProvider.class).in(Singleton.class);
    }
}

File: src/test/kotlin/at/fhhagenberg/sqe/di/MockRmiElevatorModule.kt
package at.fhhagenberg.sqe.di

import com.google.inject.AbstractModule
import com.google.inject.Key
import com.google.inject.Singleton
import sqelevator.IElevator

class MockRmiElevatorModule : AbstractModule() {
    override fun configure() {
        bind(Key.get(IElevator::class.java, RealIElevator::class.java)).toProvider(RealIElevatorMockProvider::class.java).`in`(Singleton::class.java)
    }
}